### PR TITLE
[mle] send Link Request immediately if router id was previously allocated

### PR DESCRIPTION
**Pull request by jwhui**
After upgrading to a router, the device will wait until it receives a
Route TLV from a neighboring node indicating that the new Router ID
has been allocated. In normal cases, this is not an issue since the
Leader will disseminate a new Router ID Sequence after allocating a
new Router ID. However, in some scenarios, a device may request the
same Router ID that was previously allocated and significantly delay
establishing links with neighboring routers. With this commit, a
device will immediately send a Link Request if the Router ID was
previously allocated.

### CODE REVIEW

File: src/core/thread/mle_router.cpp [hunk 1]
```diff
@@ -3866,6 +3869,17 @@ void MleRouter::HandleAddressSolicitResponse(Coap::Message *         aMessage,
...
 
+    if (isRouterIdAllocated)
+    {
+        // send Link Request immediately if Router ID was previously allocated
+        IgnoreError(SendLinkRequest(nullptr));
+    }
+    else
+    {
+        // wait to send Link Request until new Router ID has been disseminated from the Leader
+        mLinkRequestDelay = kMulticastLinkRequestDelay;
```
Comment (abtink) [inline src/core/thread/mle_router.cpp:3880]: Can we consider removing/optimizing this? We set `mLinkRequestDelay = kMulticastLinkRequestDelay` few lines below?

**Review by abtink — APPROVED** [review]
LGTM. 👍

File: src/core/thread/mle_router.cpp (hunk 1 above)
Comment (jwhui) [reply src/core/thread/mle_router.cpp:3880]: Thanks for catching that. Updated.

**Review by LuDuda — APPROVED** [review]
Thank you Jonathan! I verified this PR on my previous scenario, and now I don't see any significant delay in the communication. Also checked on Wireshark that REED indeed sends Link Request immediately after switching role back to Router again.

**Comment by jwhui** [comment]
> Thank you Jonathan! I verified this PR on my previous scenario, and now I don't see any significant delay in the communication. Also checked on Wireshark that REED indeed sends Link Request immediately after switching role back to Router again.

@LuDuda , thanks for reporting this issue and validating this fix!

**Pull request merged**